Apps/AdminWebClient/src/Server/Services/DashboardService.cs

```diff
@@ -129,16 +131,36 @@ public RequestResult<IEnumerable<MessagingVerificationModel>> GetMessageVerifica
                     dbResult = this.messagingVerificationDelegate.GetUserMessageVerifications(Database.Constants.UserQueryType.SMS, queryString);
                     break;
                 case UserQueryType.HDID:
-                    dbResult = this.messagingVerificationDelegate.GetUserMessageVerifications(Database.Constants.UserQueryType.HDID, queryString);
+                    RequestResult<PatientModel> patientResultHdid = Task.Run(async () => await this.patientService.GetPatient(queryString).ConfigureAwait(true)).Result; 
+                    if (patientResultHdid.ResultStatus == ResultType.Success && patientResultHdid.ResourcePayload != null)
+                    {
+                        phn = patientResultHdid.ResourcePayload.PersonalHealthNumber;
+                        dbResult = this.messagingVerificationDelegate.GetUserMessageVerifications(Database.Constants.UserQueryType.HDID, queryString);
+                    }
+                    else
+                    {
+                        retVal.ResultError = patientResultHdid.ResultError;
+                    }
+
                     break;
             }
 
             if (dbResult != null && dbResult.Status == Database.Constants.DBStatusCode.Read)
             {
                 retVal.ResultStatus = ResultType.Success;
+                List<MessagingVerificationModel> results = new();
                 if (dbResult.Payload != null)
                 {
-                    retVal.ResourcePayload = dbResult.Payload.Select(MessagingVerificationModel.CreateFromDbModel);
+                    results.AddRange(dbResult.Payload.Select(MessagingVerificationModel.CreateFromDbModel));
+                    if (queryType == UserQueryType.HDID || queryType == UserQueryType.PHN)
+                    {
+                        foreach (var item in results)
+                        {
+                            item.PersonalHealthNumber = phn;
+                        }
+                    }
```

Comment:
Nit: I think it might be cleaner to add the PHN as a parameter to CreateFromDbModel.

Reply:
@MikeLyttle I don't want to change the database model and to get the PHN is I need to call the service. But CreateFromDbModel should not have any service call.?

Reply:
Since we're only populating the field for a portion of the query types, it'll be easier to leave this as-is.